axiom/util_matcher.c

```diff
@@ -31,38 +38,46 @@ void nr_matcher_destroy(nr_matcher_t** matcher_ptr) {
   nr_realfree((void**)matcher_ptr);
 }
 
-bool nr_matcher_add_prefix(nr_matcher_t* matcher, const char* prefix) {
-  size_t i;
-  char* prefix_lc;
-  size_t prefix_len;
+bool nr_matcher_add_prefix(nr_matcher_t* matcher, const char* str) {
+  int i;
+  matcher_prefix* prefix;
 
-  if (NULL == matcher || NULL == prefix) {
+  if (NULL == matcher || NULL == str) {
     return false;
   }
 
-  prefix_len = nr_strlen(prefix);
-  while (prefix_len > 0 && '/' == prefix[prefix_len - 1]) {
-    prefix_len--;
+  prefix = nr_calloc(1, sizeof(matcher_prefix));
+  prefix->len = nr_strlen(str);
+  while (prefix->len > 0 && '/' == str[prefix->len - 1]) {
+    prefix->len--;
   }
 
-  prefix_lc = nr_malloc(prefix_len + 2);
-  for (i = 0; i < prefix_len; i++) {
-    prefix_lc[i] = nr_tolower(prefix[i]);
+  prefix->len += 1;
```

Comment:
Why the addition of this increment to len? It looks like the previous logic could be carried over the same without this line

Reply:
The `'+1'` is for the ending `'/'`. Previous code did `nr_malloc(prefix_len+2)` on line 48, which wasn't well explained but I managed to figure it out: +2 = +1 for the `'/'` +1 for `'\0'`. New code is still not explained very well, but does the same thing: `prefix->len += 1;` on line 55 is for the `'/'`, whereas `nr_malloc(prefix->len+1)` on line 56 does +1 for the `'\0'`